src/js/api/core-bundle-loader.js

```diff
@@ -0,0 +1,135 @@
+            const source = sources[i];
+            const providersManager = model.getProviders();
+            for (let j = 0; j < SupportsMatrix.length; j++) {
+                const provider = SupportsMatrix[j];
+                if (providersManager.providerSupports(provider, source)) {
+                    return (provider.name === providerName);
+                }
+            }
+        }
+    }
+    return false;
+}
+
+function loadControlsPolyfillHtml5Bundle() {
+    const loadPromise = require.ensure([
+        'controller/controller',
+        'view/controls/controls',
+        'intersection-observer',
+        'providers/html5'
+    ], function (require) {
+        // These modules should be required in this order
+        require('intersection-observer');
+        const CoreMixin = require('controller/controller').default;
+        ControlsLoader.controls = require('view/controls/controls').default;
+        registerProvider(require('providers/html5').default);
+        return CoreMixin;
+    }, chunkLoadErrorHandler(SETUP_ERROR_LOADING_CORE_JS + 105), 'jwplayer.core.controls.polyfills.html5');
+    bundleContainsProviders.html5 = loadPromise;
+    return loadPromise;
+}
+
+function loadControlsHtml5Bundle() {
+    const loadPromise = require.ensure([
+        'controller/controller',
+        'view/controls/controls',
+        'providers/html5'
+    ], function (require) {
+        const CoreMixin = require('controller/controller').default;
+        ControlsLoader.controls = require('view/controls/controls').default;
+        registerProvider(require('providers/html5').default);
+        return CoreMixin;
+    }, chunkLoadErrorHandler(SETUP_ERROR_LOADING_CORE_JS + 104), 'jwplayer.core.controls.html5');
+    bundleContainsProviders.html5 = loadPromise;
+    return loadPromise;
+}
+
+function loadControlsPolyfillBundle() {
+    return require.ensure([
+        'controller/controller',
+        'view/controls/controls',
+        'intersection-observer'
+    ], function (require) {
+        require('intersection-observer');
+        const CoreMixin = require('controller/controller').default;
+        ControlsLoader.controls = require('view/controls/controls').default;
+        return CoreMixin;
+    }, chunkLoadErrorHandler(SETUP_ERROR_LOADING_CORE_JS + 103), 'jwplayer.core.controls.polyfills');
+}
+
+function loadControlsBundle() {
+    return require.ensure([
+        'controller/controller',
+        'view/controls/controls'
+    ], function (require) {
+        const CoreMixin = require('controller/controller').default;
+        ControlsLoader.controls = require('view/controls/controls').default;
+        return CoreMixin;
+    }, chunkLoadErrorHandler(SETUP_ERROR_LOADING_CORE_JS + 102), 'jwplayer.core.controls');
+}
+
+function loadWebCore() {
+    return loadIntersectionObserverIfNeeded().then(loadCore);
+}
+
+function loadIntersectionObserverIfNeeded() {
+    if (requiresPolyfills()) {
```

Comment:
nitpick but thoughts on `requiresIntersectionObserverPolyfills` instead for function name?

Reply:
I'm not refactoring names or logic in this module. This change is only to separate 
`require.ensure(['controller/controller`])` or "core.js" used in headless from the other bundles not output in bin-headless.

Reply:
Sounds good